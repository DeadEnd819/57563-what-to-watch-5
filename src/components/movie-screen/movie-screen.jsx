import React, {Fragment} from "react";
import PropTypes from "prop-types";
import {Link} from "react-router-dom";
import Header from "../header/header";
import MovieList from "../movie-list/movie-list";
import Tabs from "../tabs/tabs";
import withTabs from "../../hocs/with-tabs";
import {FilmScreenType} from "../../prop-types/prop-types";
import {MORE_MOVIE_COUNT} from "../../const";
import {getFilmsByGenre} from "../../utils";

const TabsWrapped = withTabs(Tabs);

const MovieScreen = (props) => {
  const {films, film, reviews, onPlayButtonClick} = props;
  const {id, name, posterImage, backgroundImage, genre, released} = film;
  const similarFilm = getFilmsByGenre(films, genre);

  return <Fragment>
    <section className="movie-card movie-card--full">
      <div className="movie-card__hero">
        <div className="movie-card__bg">
          <img src={backgroundImage} alt={name}/>
        </div>

        <h1 className="visually-hidden">WTW</h1>

        <Header />

        <div className="movie-card__wrap">
          <div className="movie-card__desc">
            <h2 className="movie-card__title">{name}</h2>
            <p className="movie-card__meta">
              <span className="movie-card__genre">{genre}</span>
              <span className="movie-card__year">{released}</span>
            </p>

            <div className="movie-card__buttons">
              <button className="btn btn--play movie-card__button" type="button" onClick={() => onPlayButtonClick(id)}>
                <svg viewBox="0 0 19 19" width="19" height="19">
                  <use xlinkHref="#play-s"></use>
                </svg>
                <span>Play</span>
              </button>
              <button className="btn btn--list movie-card__button" type="button">
                <svg viewBox="0 0 19 20" width="19" height="20">
                  <use xlinkHref="#add"></use>
                </svg>
                <span>My list</span>
              </button>
              <Link to={`/films/${id}/review`} className="btn movie-card__button">Add review</Link>
            </div>
          </div>
        </div>
      </div>

      <div className="movie-card__wrap movie-card__translate-top">
        <div className="movie-card__info">
          <div className="movie-card__poster movie-card__poster--big">
            <img src={posterImage} alt={`${name} poster`} width="218" height="327"/>
          </div>

          <div className="movie-card__desc">

            <TabsWrapped film={film} reviews={reviews} />

          </div>
        </div>
      </div>
    </section>

    <div className="page-content">
      <section className="catalog catalog--like-this">
        <h2 className="catalog__title">More like this</h2>

        <MovieList films={similarFilm} showCount={MORE_MOVIE_COUNT} />

      </section>

      <footer className="page-footer">
        <div className="logo">
          <Link to='/' className="logo__link logo__link--light">
            <span className="logo__letter logo__letter--1">W</span>
            <span className="logo__letter logo__letter--2">T</span>
            <span className="logo__letter logo__letter--3">W</span>
          </Link>
        </div>

        <div className="copyright">
          <p>Â© 2019 What to watch Ltd.</p>
        </div>
      </footer>
    </div>
  </Fragment>;
};

MovieScreen.propTypes = {
  film: FilmScreenType.isRequired,
  films: PropTypes.array.isRequired,
  reviews: PropTypes.array.isRequired,
  onPlayButtonClick: PropTypes.func.isRequired,
};

export default MovieScreen;
