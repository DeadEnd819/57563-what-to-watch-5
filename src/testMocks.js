export const promoFilm = {
  id: 1,
  name: `The Grand Budapest Hotel`,
  posterImage: `the-grand-budapest-hotel-poster.jpg`,
  backgroundImage: `bg-the-grand-budapest-hotel.jpg`,
  backgroundColor: `#ffffff`,
  genre: `Comedy`,
  released: 2014,
  isFavorite: false,
};

export const film = {
  id: 1,
  name: `The Grand Budapest Hotel`,
  posterImage: `the-grand-budapest-hotel-poster.jpg`,
  previewImage: `bg-the-grand-budapest-hotel.jpg`,
  backgroundImage: `bg-the-grand-budapest-hotel.jpg`,
  backgroundColor: `#ffffff`,
  previewVideoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.
  Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
  rating: 8.9,
  scoresCount: 240,
  director: `Wes Andreson`,
  starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`, `Saoirse Ronan`],
  runTime: 99,
  genre: `Comedy`,
  released: 2014,
  isFavorite: true,
};

export const reviewOne = {
  id: 1,
  user: {
    id: 4,
    name: `Kate Muir`,
  },
  rating: 8.9,
  comment: `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.`,
  date: `2019-05-08T14:13:56.569Z`,
};

export const reviewTwo = {
  id: 2,
  user: {
    id: 5,
    name: `Paula Fleri-Soler`,
  },
  rating: 9.1,
  comment: `The mannered, madcap proceedings are often delightful, occasionally silly, and here and there, gruesome and/or heartbreaking.`,
  date: `2019-08-08T17:40:56.569Z`,
};

export const reviews = [reviewOne, reviewTwo];

export const genresList = [
  `Action`,
  `Adventure`,
  `Comedy`,
  `Crime`,
  `Drama`,
  `Fantasy`,
  `Thriller`
];

export const state = {
  DATA: {
    films: [film, film, film, film, film, film, film],
    promo: promoFilm,
    currentFilm: film,
    currentReviews: reviews,
  },
  FILTER: {
    genre: `Drama`,
    showFilmsCount: 8,
  },
  USER: {
    name: `name`,
    avatar: `avatar.jpg`,
    status: `NOT_AUTHORIZED`,
    reviewStatus: `REVIEW_NOT_UPDATED`,
  },
};
