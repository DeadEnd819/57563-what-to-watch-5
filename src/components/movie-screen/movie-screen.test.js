import React from "react";
import ShallowRenderer from "react-test-renderer/shallow";
import {MovieScreen} from "./movie-screen";
import {film, reviews} from "../../testMocks";

jest.mock(`../header/header.jsx`, () => `Header`);
jest.mock(`../footer/footer.jsx`, () => `Footer`);
jest.mock(`../movie-menu/movie-menu.jsx`, () => `MovieMenu`);
jest.mock(`../movie-list/movie-list.jsx`, () => `MovieList`);
jest.mock(`../tabs/tabs.jsx`, () => `Tabs`);

const renderer = new ShallowRenderer();
const noop = ()=>{};

describe(`Render MovieScreen`, () => {
  it(`Should MovieScreen isUserLogged render correctly`, () => {
    renderer.render(
        <MovieScreen
          id={`2`}
          isUserLogged={false}
          currentFilm={film}
          similarFilms={[film, film, film, film]}
          reviews={reviews}
          loadDataFilm={noop}
        />
    );

    const tree = renderer.getRenderOutput();

    expect(tree).toMatchSnapshot();
  });

  it(`Should MovieScreen render correctly`, () => {
    renderer.render(
        <MovieScreen
          id={`2`}
          isUserLogged={false}
          currentFilm={film}
          similarFilms={[film, film, film, film]}
          reviews={reviews}
          loadDataFilm={noop}
        />
    );

    const tree = renderer.getRenderOutput();

    expect(tree).toMatchSnapshot();
  });
});


