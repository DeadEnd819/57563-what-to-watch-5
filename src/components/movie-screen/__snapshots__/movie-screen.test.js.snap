// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render MovieScreen Should MovieScreen isUserLogged render correctly 1`] = `
<React.Fragment>
  <section
    className="movie-card movie-card--full"
    style={
      Object {
        "backgroundColor": "#ffffff",
      }
    }
  >
    <div
      className="movie-card__hero"
    >
      <div
        className="movie-card__bg"
      >
        <img
          alt="The Grand Budapest Hotel"
          src="bg-the-grand-budapest-hotel.jpg"
        />
      </div>
      <h1
        className="visually-hidden"
      >
        WTW
      </h1>
      <Header
        className="movie-card__head"
      />
      <div
        className="movie-card__wrap"
      >
        <MovieMenu
          film={
            Object {
              "backgroundColor": "#ffffff",
              "backgroundImage": "bg-the-grand-budapest-hotel.jpg",
              "description": "In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.
  Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.",
              "director": "Wes Andreson",
              "genre": "Comedy",
              "id": 1,
              "isFavorite": true,
              "name": "The Grand Budapest Hotel",
              "posterImage": "the-grand-budapest-hotel-poster.jpg",
              "previewImage": "bg-the-grand-budapest-hotel.jpg",
              "previewVideoLink": "https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4",
              "rating": 8.9,
              "released": 2014,
              "runTime": 99,
              "scoresCount": 240,
              "starring": Array [
                "Bill Murray",
                "Edward Norton",
                "Jude Law",
                "Willem Dafoe",
                "Saoirse Ronan",
              ],
            }
          }
          isUserLogged={false}
        />
      </div>
    </div>
    <div
      className="movie-card__wrap movie-card__translate-top"
    >
      <div
        className="movie-card__info"
      >
        <div
          className="movie-card__poster movie-card__poster--big"
        >
          <img
            alt="The Grand Budapest Hotel poster"
            height="327"
            src="the-grand-budapest-hotel-poster.jpg"
            width="218"
          />
        </div>
        <div
          className="movie-card__desc"
        >
          <Memo
            film={
              Object {
                "backgroundColor": "#ffffff",
                "backgroundImage": "bg-the-grand-budapest-hotel.jpg",
                "description": "In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.
  Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.",
                "director": "Wes Andreson",
                "genre": "Comedy",
                "id": 1,
                "isFavorite": true,
                "name": "The Grand Budapest Hotel",
                "posterImage": "the-grand-budapest-hotel-poster.jpg",
                "previewImage": "bg-the-grand-budapest-hotel.jpg",
                "previewVideoLink": "https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4",
                "rating": 8.9,
                "released": 2014,
                "runTime": 99,
                "scoresCount": 240,
                "starring": Array [
                  "Bill Murray",
                  "Edward Norton",
                  "Jude Law",
                  "Willem Dafoe",
                  "Saoirse Ronan",
                ],
              }
            }
            reviews={
              Array [
                Object {
                  "comment": "Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.",
                  "date": "2019-05-08T14:13:56.569Z",
                  "id": 1,
                  "rating": 8.9,
                  "user": Object {
                    "id": 4,
                    "name": "Kate Muir",
                  },
                },
                Object {
                  "comment": "The mannered, madcap proceedings are often delightful, occasionally silly, and here and there, gruesome and/or heartbreaking.",
                  "date": "2019-08-08T17:40:56.569Z",
                  "id": 2,
                  "rating": 9.1,
                  "user": Object {
                    "id": 5,
                    "name": "Paula Fleri-Soler",
                  },
                },
              ]
            }
          />
        </div>
      </div>
    </div>
  </section>
  <div
    className="page-content"
  >
    <section
      className="catalog catalog--like-this"
    >
      <h2
        className="catalog__title"
      >
        More like this
      </h2>
      <MovieList
        films={
          Array [
            Object {
              "backgroundColor": "#ffffff",
              "backgroundImage": "bg-the-grand-budapest-hotel.jpg",
              "description": "In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.
  Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.",
              "director": "Wes Andreson",
              "genre": "Comedy",
              "id": 1,
              "isFavorite": true,
              "name": "The Grand Budapest Hotel",
              "posterImage": "the-grand-budapest-hotel-poster.jpg",
              "previewImage": "bg-the-grand-budapest-hotel.jpg",
              "previewVideoLink": "https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4",
              "rating": 8.9,
              "released": 2014,
              "runTime": 99,
              "scoresCount": 240,
              "starring": Array [
                "Bill Murray",
                "Edward Norton",
                "Jude Law",
                "Willem Dafoe",
                "Saoirse Ronan",
              ],
            },
            Object {
              "backgroundColor": "#ffffff",
              "backgroundImage": "bg-the-grand-budapest-hotel.jpg",
              "description": "In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.
  Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.",
              "director": "Wes Andreson",
              "genre": "Comedy",
              "id": 1,
              "isFavorite": true,
              "name": "The Grand Budapest Hotel",
              "posterImage": "the-grand-budapest-hotel-poster.jpg",
              "previewImage": "bg-the-grand-budapest-hotel.jpg",
              "previewVideoLink": "https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4",
              "rating": 8.9,
              "released": 2014,
              "runTime": 99,
              "scoresCount": 240,
              "starring": Array [
                "Bill Murray",
                "Edward Norton",
                "Jude Law",
                "Willem Dafoe",
                "Saoirse Ronan",
              ],
            },
            Object {
              "backgroundColor": "#ffffff",
              "backgroundImage": "bg-the-grand-budapest-hotel.jpg",
              "description": "In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.
  Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.",
              "director": "Wes Andreson",
              "genre": "Comedy",
              "id": 1,
              "isFavorite": true,
              "name": "The Grand Budapest Hotel",
              "posterImage": "the-grand-budapest-hotel-poster.jpg",
              "previewImage": "bg-the-grand-budapest-hotel.jpg",
              "previewVideoLink": "https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4",
              "rating": 8.9,
              "released": 2014,
              "runTime": 99,
              "scoresCount": 240,
              "starring": Array [
                "Bill Murray",
                "Edward Norton",
                "Jude Law",
                "Willem Dafoe",
                "Saoirse Ronan",
              ],
            },
            Object {
              "backgroundColor": "#ffffff",
              "backgroundImage": "bg-the-grand-budapest-hotel.jpg",
              "description": "In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.
  Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.",
              "director": "Wes Andreson",
              "genre": "Comedy",
              "id": 1,
              "isFavorite": true,
              "name": "The Grand Budapest Hotel",
              "posterImage": "the-grand-budapest-hotel-poster.jpg",
              "previewImage": "bg-the-grand-budapest-hotel.jpg",
              "previewVideoLink": "https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4",
              "rating": 8.9,
              "released": 2014,
              "runTime": 99,
              "scoresCount": 240,
              "starring": Array [
                "Bill Murray",
                "Edward Norton",
                "Jude Law",
                "Willem Dafoe",
                "Saoirse Ronan",
              ],
            },
          ]
        }
        showCount={4}
      />
    </section>
    <Footer />
  </div>
</React.Fragment>
`;

exports[`Render MovieScreen Should MovieScreen render correctly 1`] = `
<React.Fragment>
  <section
    className="movie-card movie-card--full"
    style={
      Object {
        "backgroundColor": "#ffffff",
      }
    }
  >
    <div
      className="movie-card__hero"
    >
      <div
        className="movie-card__bg"
      >
        <img
          alt="The Grand Budapest Hotel"
          src="bg-the-grand-budapest-hotel.jpg"
        />
      </div>
      <h1
        className="visually-hidden"
      >
        WTW
      </h1>
      <Header
        className="movie-card__head"
      />
      <div
        className="movie-card__wrap"
      >
        <MovieMenu
          film={
            Object {
              "backgroundColor": "#ffffff",
              "backgroundImage": "bg-the-grand-budapest-hotel.jpg",
              "description": "In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.
  Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.",
              "director": "Wes Andreson",
              "genre": "Comedy",
              "id": 1,
              "isFavorite": true,
              "name": "The Grand Budapest Hotel",
              "posterImage": "the-grand-budapest-hotel-poster.jpg",
              "previewImage": "bg-the-grand-budapest-hotel.jpg",
              "previewVideoLink": "https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4",
              "rating": 8.9,
              "released": 2014,
              "runTime": 99,
              "scoresCount": 240,
              "starring": Array [
                "Bill Murray",
                "Edward Norton",
                "Jude Law",
                "Willem Dafoe",
                "Saoirse Ronan",
              ],
            }
          }
          isUserLogged={false}
        />
      </div>
    </div>
    <div
      className="movie-card__wrap movie-card__translate-top"
    >
      <div
        className="movie-card__info"
      >
        <div
          className="movie-card__poster movie-card__poster--big"
        >
          <img
            alt="The Grand Budapest Hotel poster"
            height="327"
            src="the-grand-budapest-hotel-poster.jpg"
            width="218"
          />
        </div>
        <div
          className="movie-card__desc"
        >
          <Memo
            film={
              Object {
                "backgroundColor": "#ffffff",
                "backgroundImage": "bg-the-grand-budapest-hotel.jpg",
                "description": "In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.
  Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.",
                "director": "Wes Andreson",
                "genre": "Comedy",
                "id": 1,
                "isFavorite": true,
                "name": "The Grand Budapest Hotel",
                "posterImage": "the-grand-budapest-hotel-poster.jpg",
                "previewImage": "bg-the-grand-budapest-hotel.jpg",
                "previewVideoLink": "https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4",
                "rating": 8.9,
                "released": 2014,
                "runTime": 99,
                "scoresCount": 240,
                "starring": Array [
                  "Bill Murray",
                  "Edward Norton",
                  "Jude Law",
                  "Willem Dafoe",
                  "Saoirse Ronan",
                ],
              }
            }
            reviews={
              Array [
                Object {
                  "comment": "Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.",
                  "date": "2019-05-08T14:13:56.569Z",
                  "id": 1,
                  "rating": 8.9,
                  "user": Object {
                    "id": 4,
                    "name": "Kate Muir",
                  },
                },
                Object {
                  "comment": "The mannered, madcap proceedings are often delightful, occasionally silly, and here and there, gruesome and/or heartbreaking.",
                  "date": "2019-08-08T17:40:56.569Z",
                  "id": 2,
                  "rating": 9.1,
                  "user": Object {
                    "id": 5,
                    "name": "Paula Fleri-Soler",
                  },
                },
              ]
            }
          />
        </div>
      </div>
    </div>
  </section>
  <div
    className="page-content"
  >
    <section
      className="catalog catalog--like-this"
    >
      <h2
        className="catalog__title"
      >
        More like this
      </h2>
      <MovieList
        films={
          Array [
            Object {
              "backgroundColor": "#ffffff",
              "backgroundImage": "bg-the-grand-budapest-hotel.jpg",
              "description": "In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.
  Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.",
              "director": "Wes Andreson",
              "genre": "Comedy",
              "id": 1,
              "isFavorite": true,
              "name": "The Grand Budapest Hotel",
              "posterImage": "the-grand-budapest-hotel-poster.jpg",
              "previewImage": "bg-the-grand-budapest-hotel.jpg",
              "previewVideoLink": "https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4",
              "rating": 8.9,
              "released": 2014,
              "runTime": 99,
              "scoresCount": 240,
              "starring": Array [
                "Bill Murray",
                "Edward Norton",
                "Jude Law",
                "Willem Dafoe",
                "Saoirse Ronan",
              ],
            },
            Object {
              "backgroundColor": "#ffffff",
              "backgroundImage": "bg-the-grand-budapest-hotel.jpg",
              "description": "In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.
  Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.",
              "director": "Wes Andreson",
              "genre": "Comedy",
              "id": 1,
              "isFavorite": true,
              "name": "The Grand Budapest Hotel",
              "posterImage": "the-grand-budapest-hotel-poster.jpg",
              "previewImage": "bg-the-grand-budapest-hotel.jpg",
              "previewVideoLink": "https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4",
              "rating": 8.9,
              "released": 2014,
              "runTime": 99,
              "scoresCount": 240,
              "starring": Array [
                "Bill Murray",
                "Edward Norton",
                "Jude Law",
                "Willem Dafoe",
                "Saoirse Ronan",
              ],
            },
            Object {
              "backgroundColor": "#ffffff",
              "backgroundImage": "bg-the-grand-budapest-hotel.jpg",
              "description": "In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.
  Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.",
              "director": "Wes Andreson",
              "genre": "Comedy",
              "id": 1,
              "isFavorite": true,
              "name": "The Grand Budapest Hotel",
              "posterImage": "the-grand-budapest-hotel-poster.jpg",
              "previewImage": "bg-the-grand-budapest-hotel.jpg",
              "previewVideoLink": "https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4",
              "rating": 8.9,
              "released": 2014,
              "runTime": 99,
              "scoresCount": 240,
              "starring": Array [
                "Bill Murray",
                "Edward Norton",
                "Jude Law",
                "Willem Dafoe",
                "Saoirse Ronan",
              ],
            },
            Object {
              "backgroundColor": "#ffffff",
              "backgroundImage": "bg-the-grand-budapest-hotel.jpg",
              "description": "In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.
  Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.",
              "director": "Wes Andreson",
              "genre": "Comedy",
              "id": 1,
              "isFavorite": true,
              "name": "The Grand Budapest Hotel",
              "posterImage": "the-grand-budapest-hotel-poster.jpg",
              "previewImage": "bg-the-grand-budapest-hotel.jpg",
              "previewVideoLink": "https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4",
              "rating": 8.9,
              "released": 2014,
              "runTime": 99,
              "scoresCount": 240,
              "starring": Array [
                "Bill Murray",
                "Edward Norton",
                "Jude Law",
                "Willem Dafoe",
                "Saoirse Ronan",
              ],
            },
          ]
        }
        showCount={4}
      />
    </section>
    <Footer />
  </div>
</React.Fragment>
`;
